core/src/torii/routing.rs

```diff
@@ -289,6 +289,50 @@ async fn update_metrics<W: WorldTrait>(
     Ok(())
 }
 
+/// Convert accumulated `Rejection` into appropriate `Reply`.
+#[allow(clippy::unused_async)]
+// TODO: -> Result<impl Reply, Infallible>
+pub(crate) async fn handle_rejection(rejection: Rejection) -> Result<Response, Rejection> {
+    use super::Error::*;
+
+    let err = if let Some(err) = rejection.find::<Error>() {
+        err
+    } else {
+        iroha_logger::error!(?rejection, "unhandled rejection");
+        return Ok(StatusCode::INTERNAL_SERVER_ERROR.into_response());
+    };
+
+    #[allow(clippy::match_same_arms)]
+    let response = match err {
+        Query(err) => err.clone().into_response(),
+        VersionedTransaction(err) => {
+            reply::with_status(err.to_string(), err.status_code()).into_response()
+        }
+        AcceptTransaction(_err) => return unhandled(rejection),
+        RequestPendingTransactions(_err) => return unhandled(rejection),
+        DecodeRequestPendingTransactions(err) => {
+            reply::with_status(err.to_string(), err.status_code()).into_response()
+        }
+        EncodePendingTransactions(err) => {
+            reply::with_status(err.to_string(), err.status_code()).into_response()
+        }
+        TxTooBig => return unhandled(rejection),
+        Config(_err) => return unhandled(rejection),
+        PushIntoQueue(_err) => return unhandled(rejection),
+        Status(_err) => return unhandled(rejection),
+        ConfigurationReload(_err) => return unhandled(rejection),
+        Prometheus(_err) => return unhandled(rejection),
+    };
+
+    Ok(response)
+}
+
+// TODO: Remove this. Handle all the `Error` cases in `handle_rejection`
+fn unhandled(rejection: Rejection) -> Result<Response, Rejection> {
+    iroha_logger::error!(?rejection, "unhandled rejection");
```

Comment:
Logging rejections is a good idea. Not sure if `error` is the right level though. I'd suggest reserving `error` only for things that can break the Peer itself, while rejections are `warning`s.